test/test_sparse.py

```diff
@@ -1080,6 +1080,49 @@ def test_shape(di, dj, dk, nnz):
         test_shape(1000, 0, 100, 0)
         test_shape(1000, 100, 0, 0)
 
+    @cpu_only
+    def test_sspaddmm(self):
+
+        def test_shape(di, dj, dk, nnz):
+            x = self._gen_sparse(2, nnz, [di, dj])[0]
+            t = self._gen_sparse(2, nnz, [di, dk])[0]
+            y = torch.randn(dj, dk)
+            alpha = random.random()
+            beta = random.random()
+
+            res = t.sspaddmm(x, y, beta=beta, alpha=alpha)
+            expected = torch.addmm(self.safeToDense(t), self.safeToDense(x), y, beta=beta, alpha=alpha)
+            self.assertEqual(self.safeToDense(res), expected)
+
+            res = t.sspaddmm(x, y)
+            expected = torch.addmm(self.safeToDense(t), self.safeToDense(x), y)
+            self.assertEqual(self.safeToDense(res), expected)
+
+        test_shape(7, 5, 3, 20)
+        test_shape(1000, 100, 100, 20)
+        test_shape(3000, 64, 300, 20)
+        test_shape(0, 100, 100, 0)
+        test_shape(1000, 0, 100, 0)
+        test_shape(1000, 100, 0, 0)
+
+        # Test code from issue https://github.com/pytorch/pytorch/issues/45113
+        batch_size, input_size, hidden_size = 5, 3, 7
+        # Create uncoalesced sparse tensors:
+        weight = torch.randn(hidden_size, 1).to_sparse()
```

Comment:
To really test what you want to test you have to prepare weights like in the original issue, 
```
weight = torch.randn(hidden_size, input_size).to_sparse()
```
Then, due to bad reasons the resulting tensor would be both coalesced (`.is_coalesced` will be `True`, and `.coalesce` would not touch indices), and indices would still be discontiguous. 
However, a better solution is to introduce an invariant that tensor that is `coalesced` always has contiguous `indices` and `values` (which is almost true today). This hotfix with a test that is actually testing it will do for now though.

Reply:
@ngimel , introducing (or not) a new invariant "coalesced implies contiguous" seems to be related to https://github.com/pytorch/pytorch/issues/12633 that suggests the approach used in this PR.

Reply:
I don't mind approach used in this PR, but test should be testing a fix, which it does not now.

Reply:
@pearu @ngimel thanks for the comments, I'll update the test to check the fix.